File: UBiiiii/Week5/달리기경주/Solution.java
package 달리기경주;

import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        String[] answer = new String[players.length];
        Map<String, Integer> map = new HashMap<>();
        int i = 0;
        for(String player : players) {
            map.put(player, i);
            answer[i++] = player;
        }
        
        for(String player: callings) {
            int idx = map.get(player);
            
            if(idx==0) continue;
            String tmp = answer[idx-1];
            answer[idx-1] = answer[idx];
            answer[idx] = tmp;
            map.put(player, idx-1);
            map.put(tmp, idx);
        }
        return answer;
    }
}

File: UBiiiii/Week5/달리기경주/Solution.js
function solution(players, callings) {
    var answer = [];
    let position = {};
    players.forEach((v, i) => {
        position[v] = i;
    })
    callings.forEach(el => {
        const curr = position[el];
        const pre = players[curr-1];
        players[curr-1] = players[curr];
        players[curr] = pre;
        position[el] = curr-1;
        position[pre] = curr;
    });

    return players;
}